ouroboros-consensus/src/unstable-consensus-testlib/Test/Ouroboros/Consensus/ChainGenerator/BitVector.hs

```diff
@@ -162,11 +163,17 @@ fillInWindow ::
   -> g
   -> C.MVector base SlotE s S
   -> ST s (C.Var base (PreImage pol ActiveSlotE))   -- ^ the count after filling
-fillInWindow pol (SomeDensityWindow k s) g mv = do
+fillInWindow pol (SomeDensityWindow k s) g mv
+        | not (C.getCount k <= C.getCount s) =
+            error $ "fillInWindow: assertion failure: k <= s: "
+                    ++ show k ++ " <= " ++ show s
```

Comment:
Hmm, do we want to (also) make this check part of creating a `SomeDensityWindow`?

Reply:
I moved the assertion to a smart constructor.

Reply:
I undid this. Worth addressing in some other PR. I couldn't come up quickly with a solution that was obviously ideal.